CCRing.c

```diff
@@ -50,3 +51,33 @@ CCError ringAppend(CCRing* pRing, ccAudioDataType arr[], unsigned long length) {
   }
   return ccNoError;
 }
+
+
+CCError ccGenerateSin(CCRing* sinusoid, double cycles) {
+  unsigned long length = sinusoid->length;
+  double distBetweenPoints = (2*PI)/length;
+
+  ccAudioDataType dataPoints[length];
```

Comment:
On line 60 above you do not actually need to allocate a new array here - the memory we want to access is already available - it is accessible through the `*data` member of the audio ring struct. 

I believe This approach for allocating an array inside a function will not work correctly.  This is a 'dynamic' array, meaning that the length is not known at compile time. For that reason the memory for the array would need to be allocated with the `malloc` function. This may be why the code is not working as you expect. In this case we do not need to allocate a new array at all -- just insert the sine values directly into the `data` member of the sinusoid structure.